iOS Client/DevotionsController/DevotionsCoordinator.swift

```diff
@@ -89,11 +101,37 @@ class DevotionsCoordinator: NSObject, Coordinator, StoreSubscriber, UNUserNotifi
     
     // MARK: - UNNotificationDelegate
     func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+        
+        if response.actionIdentifier == UNNotificationDefaultActionIdentifier {
+            selectTodaysDevotion()
+        } else {
+            // user dismissed notification
+        }
+        
+        // notify the system that we're done
+        completionHandler()
+    }
+    
+    func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        completionHandler(UNNotificationPresentationOptions.alert)
+    }
+    
+    private func selectTodaysDevotion() {
         let devotions = devotionsStore.getDevotions()
-        let date = Date()
-        if let devotion = devotions.first(where: { $0.date == LoadedDevotion.devotionDateFormatter.string(from: date) }) {
+        MBStore.sharedStore.dispatch(LoadedDevotions(devotions: .loaded(data: devotions)))
+        
+        if let devotion = devotions.first(where: {$0.dateAsMMdd == Date().toMMddString()}) {
             MBStore.sharedStore.dispatch(SelectedDevotion(devotion: devotion))
         }
-        MBStore.sharedStore.dispatch(LoadedDevotions(devotions: .loaded(data: devotions)))
+    }
+}
+
+extension Date
+{
+    func toMMddString() -> String
+    {
+        let dateFormatter = DateFormatter()
```

Comment:
Apparently DateFormatters are actually expensive to initialize so I would make this a static or lazy property. Don't think it really will have an effect but we are looping over all the devotions and initializing a date formatter every time so it might.

Reply:
reasonable